src/RazorSdk/SourceGenerators/RazorSourceGenerationContext.cs

```diff
@@ -0,0 +1,197 @@
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators
+{
+    internal class RazorSourceGenerationContext
+    {
+        public string RootNamespace { get; private set; }
+
+        public IReadOnlyList<RazorInputItem> RazorFiles { get; private set; }
+
+        public IReadOnlyList<RazorInputItem> CshtmlFiles { get; private set; }
+
+        public VirtualRazorProjectFileSystem FileSystem { get; private set; }
+
+        public RazorConfiguration Configuration { get; private set; }
+
+        public bool DesignTimeBuild { get; private set; }
+
+        public string RefsTagHelperOutputCachePath { get; private set; }
+
+        /// <summary>
+        /// Gets a flag that determines if the source generator waits for the debugger to attach.
+        /// <para>
+        /// To configure this using MSBuild, use the <c>_RazorSourceGeneratorDebug</c> property.
+        /// For instance <c>dotnet msbuild /p:_RazorSourceGeneratorDebug=true</c>
+        /// </para>
+        /// </summary>
+        public bool WaitForDebugger { get; private set; }
+
+        /// <summary>
+        /// Gets a flag that determines if generated output is to be written to disk.
+        /// Primarily meant for tests and debugging.
+        /// <para>
+        /// To configure this using MSBuild, use the <c>_RazorSourceGeneratorWriteGeneratedOutput</c> property.
+        /// For instance <c>dotnet msbuild /p:_RazorSourceGeneratorWriteGeneratedOutput=true</c>
+        /// </para>
+        /// </summary>
+        public bool WriteGeneratedContent { get; private set; }
+
+        /// <summary>
+        /// Gets a flag that determine if the source generator should log verbose messages.
+        /// </summary>
+        /// <para>
+        /// To configure this using MSBuild, use the <c>_RazorSourceGeneratorLog</c> property.
+        /// For instance <c>dotnet msbuild /p:_RazorSourceGeneratorLog=true</c>
+        /// </para>
+        public bool EnableLogging { get; private set; }
+
+        public static RazorSourceGenerationContext Create(GeneratorExecutionContext context)
+        {
+            var globalOptions = context.AnalyzerConfigOptions.GlobalOptions;
+
+            if (!globalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace))
+            {
+                rootNamespace = "ASP";
+            }
+
+            globalOptions.TryGetValue("build_property.DesignTimeBuild", out var designTimeBuild);
+            if (!globalOptions.TryGetValue("build_property._RazorReferenceAssemblyTagHelpersOutputPath", out var refsTagHelperOutputCachePath))
+            {
+                throw new InvalidOperationException("_RazorReferenceAssemblyTagHelpersOutputPath is not specified.");
```

Comment:
Rather than throwing consider adding a diagnostic with error severity. That will produce a better customer experience here